src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -504,6 +504,24 @@ internal static class Assembly
                 .CreateDelegate<Func<System.Reflection.Assembly, string, bool, bool, Type>>();
         }
 
+        internal static class AssemblyName
+        {
+            private const string TypeName = "System.Reflection.AssemblyName";
+
+            internal static readonly Type Type = ReflectionUtilities.GetTypeFromEither(
+                contractName: $"{TypeName}, {CoreNames.System_Reflection}",
+                desktopName: TypeName);
+
+            private static readonly PropertyInfo CodeBase = Type
+                .GetTypeInfo()
+                .GetDeclaredProperty(nameof(CodeBase));
```

Comment:
I looked at our current analyzer assembly loaders and all of them are implemented in the non-portable layer - which makes sense as all the analyzer hosts are also in the non-portable layer. I am going to move this particular loader to the VS layer and we can figure out later how to load analyzer assemblies in the portable layer, when such a host exists.